analyzers/tests/SonarAnalyzer.UnitTest/TestCases/MagicNumberShouldNotBeUsed.cs

```diff
@@ -53,26 +116,53 @@ public class WrongUseCases
     {
         public static double FooProp
         {
-            get { return Math.Sqrt(4); } // Noncompliant
+            get { return Math.Sqrt(4); } // Noncompliant {{Assign this magic number '4' to a well-named variable/constant, and use that instead.}}
+//                                 ^
         }
 
-        public WrongUseCases()
+        public WrongUseCases(int x, int y) { }
+
+        public WrongUseCases(List<int> list, string s, FooBar foo)
         {
-            Console.WriteLine(12); // Noncompliant
+            Console.WriteLine(12); // Noncompliant {{Assign this magic number '12' to a well-named variable/constant, and use that instead.}}
 
-            for (int i = 10; i < 50; i++) // Noncompliant
+            for (int i = 10; i < 50; i++) // Noncompliant {{Assign this magic number '50' to a well-named variable/constant, and use that instead.}}
             {
 
             }
 
             var array = new string[10];
-            array[5] = "test"; // Noncompliant
+            array[5] = "test"; // Noncompliant {{Assign this magic number '5' to a well-named variable/constant, and use that instead.}}
+            Foo(new int[] { 100 }); // Noncompliant
+
+            new WrongUseCases(100, Foo(200, 300)); // Noncompliant {{Assign this object creation with magic number '100' to a well-named variable/constant, and use that instead.}}
+            // Noncompliant@-1 {{Assign this magic number '200' to a well-named variable/constant, and use that instead.}}
+            // Noncompliant@-2 {{Assign this magic number '300' to a well-named variable/constant, and use that instead.}}
+
+            Foo(new IntPtr(123456)); // Noncompliant {{Assign this object creation with magic number '123456' to a well-named variable/constant, and use that instead.}}
+//              ^^^^^^^^^^^^^^^^^^
+            Foo(new int[] { 100, 200 }); // Noncompliant {{Assign this magic number '100' to a well-named variable/constant, and use that instead.}}
+            // Noncompliant@-1 {{Assign this magic number '200' to a well-named variable/constant, and use that instead.}}
+
+            var result = list.Count == 99;
+            result = list.Count < 400; // Noncompliant
+            result = s.Length == 121; // Noncompliant
+            result = foo.Size == 472; // Noncompliant
+            result = foo.baz == 4; // Noncompliant
```

Comment:
maybe for single digits should just ignore the rule when doing comparisons 

because there could be also cases like `HowMany() == 2` in code... not sure on this one